### PR TITLE
Cache MessageRegexp

### PR DESCRIPTION
**Pull request by steelbrain**
Fixes https://github.com/AtomLinter/Linter/issues/687

### CODE REVIEW

File: lib/linter.coffee [hunk 1]
```diff
@@ -2,6 +2,7 @@ fs = require 'fs'
 path = require 'path'
 {CompositeDisposable, Range, Point, BufferedProcess} = require 'atom'
 XRegExp = null
+MessageRegexp = null
```
Comment (iam4x) [inline lib/linter.coffee]: Can't we keep `MessageRegexp` on the `Linter` instance?

Comment (steelbrain) [inline lib/linter.coffee]: We can, but the linter instance isn't shared, so I find this a bit way cleaner.

**Comment by maxbrunsfeld** [comment]
I think this change caused `linter-clang` to stop working. It looks like `linter-clang` [uses a different RegExp](https://github.com/AtomLinter/linter-clang/blob/dae931d41fa8a09d4a0189202f4e5784414e924b/lib/linter-clang.coffee#L147) each time `lintFile` is called, because the RegExp is based on the filename. This caching causes the wrong RegExp to be used. I don't know if this should be considered a bug in `linter` or in `linter-clang`.

**Comment by steelbrain** [comment]
In any case, It's using the legacy API, it should upgrade to solve this issue :)

**Comment by AsaAyers** [comment]
@maxbrunsfeld If you'd like an actual fix that can be done immediately, it looks like you can clear the cache with this:

```CoffeeScript
    # add file to regex to filter output to this file,
    # need to change filename a bit to fit into regex
    delete @MessageRegexp
    @regex = filePath.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&") +
```

**Comment by maxbrunsfeld** [comment]
Thanks @AsaAyers, @steelbrain. I've just installed linter 1.1.0 for the time being.

**Pull request merged**